components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil.java

```diff
@@ -146,6 +151,12 @@ public static boolean isSignatureVerified(SignedJWT signedJWT, String jwksUri) t
         if (StringUtils.isNotBlank(jwksUri)) {
             String jwtString = signedJWT.getParsedString();
             String alg = signedJWT.getHeader().getAlgorithm().getName();
+            String clientId = getClientIdFromPayload(signedJWT.getPayload());
```

Comment:
Move to a common place

Reply:
fixed with 8094265d4abb76d2dd4435c246b499c4a4eb35cf